eclipse/che
# search username in namespace annotations

Signed-off-by: Michal Vala <mvala@redhat.com>



### What does this PR do?
Evaluates username and try to find it in annotations, instead of labels.

TODO:

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/18176

### Documentation PR
https://github.com/eclipse/che-docs/pull/1659


### How to test this PR?

1. Have user with non-DNS username (like an email)
1. Prepare namespace with labels and annotations like and create admin rolebinding for the user in that namespace:
```
metadata:
  annotations:
    che.eclipse.org/username: tester@tester.cz
  labels:
    app.kubernetes.io/component: workspace
    app.kubernetes.io/part-of: che.eclipse.org
```
3. New workspace should be created there, either with dashboard or factory

This also fixes the issue with namespace listing, which was not working at all.
1. Have user with non-DNS username (like an email)
1. Create new custom workspace from dashboard. It should work.


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
